### ISSUE TITLE
Support `@JsonFormat(shape=STRING)` for serialization of `Range<T>`

### ISSUE THREAD

**Issue by chris-dekker** [FasterXML/jackson-datatypes-collections#118]
I find the standard serialization JSON, while flexible, of `Range<T>` to be quite verbose:

```
{
  "lower_endpoint": 12,
  "lower_bound_type": "CLOSED",
  "upper_endpoint": 34,
  "upper_bound_type": "OPEN"
}
```

When manually processing the results, I prefer to shorter notation produced by, among others, the default `toString()` invocation: `[12..34)`.

Similar to `Instant`, where we can use `@JsonFormat(shape = ..)` to control (de)serialization to use the ISO format or unix milis, it would be great if we could control the _shape_ of the range this way. Keep the default as-is and only when `@JsonFormat(shape = STRING)` is provided, use the shorter notation.

Granted, this will only work for endpoints having scalar values, but that seems to cover the majority of use cases I've seen so far.

**Comment by cowtowncoder** [comment]
Sounds like a good idea for someone with time & interest to work on, and solid improvement to usability. Tagged as such.

**Comment by saydar31** [comment]
Assign it to me please,
Also I have a question: how should range without lower or upper bound look like?
(,1] or (-INF, 1], in my humble opinion second one looks better

**Comment by cowtowncoder** [comment]
@saydar31 I let you decide. But does Guava's `Range.toString()` offer guidance here?

**Comment by chris-dekker** [comment]
@saydar31 would prefer to stay consistent with the `Range.toString()` method indeed, which uses the `∞` symbol (`\u221e`)

**Comment by cowtowncoder** [comment]
Merged for inclusion in 2.17.0
